### PR TITLE
fix: Table's last column doesn't resize properly

### LINKED ISSUE #15843: [Bug]: Table's last column doesn't resize properly
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When resizing the last column of a table, it doesn't resize as expected.

https://www.loom.com/share/2fc4d376171f44d9a7604ec38c83e625

### Steps To Reproduce

1. Add a table
2. Resize the table's last column
3. Observe there is a gap at the end no matter how much you resize

### Public Sample App

_No response_

### Version

Production

### PR DESCRIPTION
**Pull request by souma-ghosh**
## Description
Last table column will now resize down to a minimum width of 150px

Fixes #15843 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by souma-ghosh** [comment]
@sbalaji1192 Since we don't have a well defined target for cypress to drag and resize a table column we are proceeding with manual testing in this

**Comment by souma-ghosh** [comment]
Dynamic Column Resizing behaviour

- Only the last column’s width is dynamically adjusted just before rendering
- The dynamic resizing effect won’t kick in if more than one column is spilling out.
- Dynamic resizing will downsize the last column to a minimum of COLUMN_MIN_WIDTH (currently 60px)
- The adjusted width of the last column is softly set i.e. this adjusted width does not reflect in columnWidthMap. Hence we are able to retain the original width of the last column be it dynamically down sized or upsized.
- Since we retain the original width of the last column the original width will take effect as soon as this column is reordered to any other position. As a result in this scenario the new last column will now experience the effect of dynamic resizing.

This explains the current behaviour of the column resizing effect. Please take a look at this before we can move further
@sbalaji1192 @dilippitchika

**Comment by sbalaji1192** [comment]
@souma-ghosh Please move on with QA. Will review it once its passes the QA cycle

**Comment by laveena-en** [comment]
Tested and verified.

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 1]
```diff
@@ -781,7 +812,9 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
       isVisiblePagination ||
       isVisibleSearch;
 
-    const { componentHeight, componentWidth } = this.getComponentDimensions();
+    // eslint-disable-next-line prefer-const
+    let { componentHeight, componentWidth } = this.getComponentDimensions();
+    componentWidth = componentWidth - 2 * WIDGET_PADDING;
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/index.tsx:817]: Add comment for 2

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 2]
```diff
@@ -158,7 +162,8 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
     let columns: ReactTableColumnProps[] = [];
     const hiddenColumns: ReactTableColumnProps[] = [];
 
-    const { componentWidth } = this.getComponentDimensions();
+    let { componentWidth } = this.getComponentDimensions();
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/index.tsx:165]: override this function in the table widget.

**Review by sbalaji1192 — APPROVED** [review]

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**